mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/plan/synchronization/IndexingPlanSynchronizationStrategyConfigurationContext.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.plan.synchronization;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+import org.hibernate.search.engine.backend.work.execution.DocumentCommitStrategy;
+import org.hibernate.search.engine.backend.work.execution.DocumentRefreshStrategy;
+import org.hibernate.search.engine.backend.work.execution.OperationSubmitter;
+import org.hibernate.search.engine.reporting.FailureHandler;
+import org.hibernate.search.mapper.pojo.work.SearchIndexingPlanExecutionReport;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+@Incubating
+public interface IndexingPlanSynchronizationStrategyConfigurationContext<E, R extends SearchIndexingPlanExecutionReport<E>> {
```

Comment:
As far as I can see, `R` is not necessary here. You could remove it and use `SearchIndexingPlanExecutionReport<E>` everywhere you currently use `R`. The only place where it could be a problem is `org.hibernate.search.mapper.orm.work.SearchIndexingPlanExecutionReport`, and you already use adapters for that one anyway.

Thinking about it, we might be able to remove `E` too: the class `EntityReference` in the standalone and ORM mapper are identical, so we could just merge them. That would however require most extensive changes, so maybe this should be done as a follow-up, in a separate PR/issue.